Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java

```diff
@@ -259,4 +243,33 @@ private PlatformGamesView getPlatformGamesView(Platform platform)
 
     return (PlatformGamesView) getSupportFragmentManager().findFragmentByTag(fragmentTag);
   }
+
+  private void setPlatformTabsAndStartGameFileCacheService()
+  {
+    PlatformPagerAdapter platformPagerAdapter = new PlatformPagerAdapter(
+            getSupportFragmentManager(), this);
+    mViewPager.setAdapter(platformPagerAdapter);
+    mViewPager.setOffscreenPageLimit(platformPagerAdapter.getCount());
+    mTabLayout.setupWithViewPager(mViewPager);
+    mTabLayout.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(mViewPager)
+    {
+      @Override
+      public void onTabSelected(@NonNull TabLayout.Tab tab)
+      {
+        super.onTabSelected(tab);
+        NativeLibrary
+                .SetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_ANDROID,
+                        SettingsFile.KEY_LAST_PLATFORM_TAB, Integer.toString(tab.getPosition()));
+      }
+    });
+
+    // Don't call NativeLibrary before DirectoryInitialization.
```

Comment:
I don't understand why there is a comment like this in this specific position. It would probably make more sense to put a comment above this function that says this function shouldn't be called before directory initialization, so as to provide documentation for callers of this code.

And isn't this code in fact potentially running `NativeConfig.GetConfig` before directory initialization is done? `DirectoryInitialization.start` is asynchronous, so all you're guaranteeing by calling that first is that directory initialization has started, not that is has finished.

Reply:
I was aware that my code was suspicious but from my testing I didn't run into any problems with DirectioryInitialization. I wanted to leave a comment to draw attention to that. But I just remembered that `AfterDirectoryInitializationRunner` exists so that should solve the concerns!